### PR TITLE
docs(examples): add url sync

### PR DESCRIPTION
**Pull request by mthuret**
uses browser history and a 700 ms threshold. 

the url sync on our examples does not cover:
* removing empty refinements
* ordering query params by alphabetical order

I can add them if you think it worth it.

### CODE REVIEW

File: docgen/src/examples/urlSync.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React, {Component} from 'react';
+import qs from 'qs';
+
+export const withUrlSync = App => class extends Component {
+  constructor() {
+    super();
+    const history = window.history;
+    this.state = {history, state: {...qs.parse(window.location.search.slice(1))}};
```
Comment (vvo) [inline docgen/src/examples/urlSync.js]: Why do you need to store window.history in the state? For convenience?

Comment (mthuret) [inline docgen/src/examples/urlSync.js]: could totally be replaced with window.history each time yeah (like location)

File: docgen/src/examples/urlSync.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    } else {
+      this.state.history.pushState(null, null, search);
+    }
+  };
+
+  createURL = state => `?${qs.stringify(state)}`;
+
+  render() {
+    return <App {...this.props}
+                state={this.state.state}
```
Comment (vvo) [inline docgen/src/examples/urlSync.js]: Maybe we could name it `this.state.searchState`?

**Comment by vvo** [comment]
LGTM, ultimately empty refinements entries could just be removed from the state I guess?

**Comment by mthuret** [comment]
Yes I was thinking the same. a new refactor to do ? :]

**Comment by vvo** [comment]
> Yes I was thinking the same. a new refactor to do ? :]

Yep in another step :)

**Pull request merged**